server/src/main/java/io/crate/expression/predicate/IsNullPredicate.java

```diff
@@ -109,32 +109,28 @@ public Query toQuery(Function function, Context context) {
             if (!ref.isNullable()) {
                 return new MatchNoDocsQuery("`x IS NULL` on column that is NOT NULL can't match");
             }
-            Query refExistsQuery = refExistsQuery(ref, context, true);
+            Query refExistsQuery = refExistsQuery(ref, context);
             return refExistsQuery == null ? null : Queries.not(refExistsQuery);
         }
         return null;
     }
 
 
     @Nullable
-    public static Query refExistsQuery(Reference ref, Context context, boolean countEmptyArrays) {
+    public static Query refExistsQuery(Reference ref, Context context) {
```

Comment:
`countEmptyArrays` can be removed because when `ref` is an array type, `countEmptyArrays` passed in is always `true`.